questdb-rs/src/ingress/mod.md

```diff
@@ -0,0 +1,232 @@
+TLS documentation](https://questdb.io/docs/operations/tls/).
+
+*Note*: QuestDB Open Source does not support TLS natively. To use TLS with
+QuestDB Open Source, use a TLS proxy such as
+[HAProxy](http://www.haproxy.org/).
+
+We support several certification authorities (sources of PKI root certificates).
+To select one, use the `tls_ca` config option. These are the supported variants:
+
+* `tls_ca=webpki_roots;` use the roots provided in the standard Rust crate
+  [webpki-roots](https://crates.io/crates/webpki-roots)
+
+* `tls_ca=os_roots;` use the OS-provided certificate store
+
+* `tls_ca=webpki_and_os_roots;` combine both of the above
+
+* `tls_roots=/path/to/root-ca.pem;` get the root certificates from the specified
+  file. Main purpose is for testing with self-signed certificates. _Note:_ this
+  automatically sets `tls_ca=pem_file`.
+
+See our notes on [how to generate a self-signed
+certificate](https://github.com/questdb/c-questdb-client/tree/main/tls_certs).
+
+* `tls_verify=unsafe_off;` tells the QuestDB client to ignore all CA roots and
+  accept any server certificate without checking. You can use it as a last
+  resort, when you weren't able to apply the above approach with a self-signed
+  certificate. You should **never use it in production** as it defeats security
+  and allows a man-in-the middle attack.
+
+## HTTP Timeouts
+
+Instead of a fixed timeout value, we use a flexible timeout that depends on the
+size of the HTTP request payload (how much data is in the buffer that you're
+flushing). You can configure it using two options:
+
+* `request_min_throughput` (bytes per second, default 100 KiB/s): divide the
+  payload size by this number to determine for how long to keep sending the
+  payload before timing out.
+* `request_timeout` (milliseconds, default 10 seconds): additional time
+  allowance to account for the fixed latency of the request-response roundtrip.
+
+Finally, the client will keep retrying the request if it experiences errors. You
+can configure the total time budget for retrying:
+
+* `retry_timeout` (milliseconds, default 10 seconds)
+
+# Usage Considerations
+
+The [Library
+considerations](https://github.com/questdb/c-questdb-client/blob/main/doc/CONSIDERATIONS.md)
+document covers these topics:
+
+* Threading
+* Differences between ILP and QuestDB Data Types
+* Data Quality
+* Client-side checks and server errors
+* Flushing
+* Disconnections, data errors and troubleshooting
+
+Here are a few more things to keep in mind.
+
+## Sequential Coupling in the Buffer API
+
+The fluent API of [`Buffer`] has sequential coupling: there's a certain order in
+which you are expected to call the methods. For example, you must write the
+symbols before the columns, and you must terminate each row by calling either
+[`at`](Buffer::at) or [`at_now`](Buffer::at_now). Refer to the [`Buffer`] doc
+for the full rules and a flowchart.
+
+## Flushing
```

Comment:
consider moving this up in the docs? IMHO grasping flushing (and error handling) is perhaps the most important concept user need to understand. even more so given the Rust client does not auto-flush.

Reply:
I'm moving it up and rephrasing a bit. But keep in mind that this issue is way more obvious in Rust, which doesn't use the `Sender` object to add data. It uses two separate objects: `Buffer` and `Sender`, so it jumps out at you that you must pass the buffer to sender to get any action.